### PR TITLE
Regex fixes

### PR DESCRIPTION
**Pull request by michaelcfanning**
- RUL: Add `SEC101/061.LooseOAuth2BearerToken` detection.
- BRK: Remove `SEC101/109.AzureContainerRegistryLegacyKey` as it is too anonymous for standalone secret detection.
- FNS: Add `UrlCredentials` to `WellKnownPatterns.UnclassifiedPotentialSecurityKeys`.
- FNS: Add `Unclassified32CharacterString` to `WellKnownPatterns.UnclassifiedPotentialSecurityKeys`. This rule locates some legacy AAD app passwords as well as legacy Azure container registry keys. 

OK, so apparently, I previously left a number of rules out of well known patterns for a variety of underlying reasons. Some patterns had syntax errors. In another case, the ACR rule collided with some aad client app detections. The URL credentials rule contained a back-tracking pattern.

All now resolved.

### CODE REVIEW

File: src/Microsoft.Security.Utilities.Core/UrlCredentials.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ public UrlCredentials()
 
         Name = "UrlCredentials";
 
-        Pattern = $"(?<=//[^:/?#\\n]+:){Url}(?=@)";
+        Pattern = @"https?:\/\/(?:[^:@]+):(?<refine>[^:@?]+)@";
```
Comment (michaelcfanning) [inline src/Microsoft.Security.Utilities.Core/UrlCredentials.cs:17]: >Pattern [](http://example.com/codeflow?start=8&length=7)

I've taken this simplification for the non-back-tracking case.

File: GeneratedRegexPatterns/PreciselyClassifiedSecurityKeys.json [hunk 2]
```diff
@@ -1,4 +1,20 @@
 [
+  {
+    "Pattern": "^(?i)(?<refine>[a-z0-9=+/]{32})$",
+    "Id": "SEC101/109",
+    "Name": "AzureContainerRegistryLegacyKey",
+    "Signatures": null,
+    "DetectionMetadata": "ObsoleteFormat, HighEntropy"
+  },
+  {
+    "Pattern": "(^|[^abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890+/-_=])(?P<secret>[abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890]{52}JQQJ9(?:9|D)[abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890][A-L][abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890]{16}[A-Za-z][abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890]{7}(?:[abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890]{2}==)?)",
```
Comment (rwoll) [inline GeneratedRegexPatterns/PreciselyClassifiedSecurityKeys.json:10]: Why does this regex both use shorthand (`[0-9a-zA-Z]`) and explicitly listing every char (`[abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890]`)?

File: src/Tests.Microsoft.Security.Utilities.Core/SecretMaskerTests.cs [hunk 3]
```diff
@@ -13,7 +13,6 @@
 
 namespace Microsoft.Security.Utilities;
 
-[TestClass, ExcludeFromCodeCoverage]
```
Comment (rwoll) [inline src/Tests.Microsoft.Security.Utilities.Core/SecretMaskerTests.cs:16]: double check you still observe these tests running in CI environments still.

File: GeneratedRegexPatterns/UnclassifiedPotentialSecurityKeys.json [hunk 4]
```diff
@@ -20,6 +30,23 @@
     ],
     "DetectionMetadata": "HighEntropy, MediumConfidence"
   },
+  {
+    "Pattern": "(?i)authorization:(\\s|%20)bearer(\\s|%20)(?<refine>[0-9a-z][abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890-_~.+\\/=]*)([^abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890-_~.+/=]|$)",
+    "Id": "SEC101/061",
+    "Name": "OAuth2BearerToken",
+    "Signatures": [
```
Comment (rwoll) [inline GeneratedRegexPatterns/UnclassifiedPotentialSecurityKeys.json:37]: How do these signatures apply to this scenario?

File: src/Microsoft.Security.Utilities.Core/UnclassifiedPotentialSecurityKeys/SEC000_003_Unclassified32CharacterString.cs [hunk 5]
```diff
@@ -49,15 +49,26 @@ public override IEnumerable<string> GenerateTruePositiveExamples()
                     yield return key;
                     break;
                 }
+
+                yield return WellKnownRegexPatterns.GenerateString($"{WellKnownRegexPatterns.Base62}.=-:[_@/*]+?", 32);
```
Comment (rwoll) [inline src/Microsoft.Security.Utilities.Core/UnclassifiedPotentialSecurityKeys/SEC000_003_Unclassified32CharacterString.cs:53]: `.=-:[_@/*]+?` is repeated a lot. Can you add an explanatory variable?

Is this the same line repeated 4 times? What's the purpose of it?

File: src/Tests.Microsoft.Security.Utilities.Core/SecretMaskerTests.cs [hunk 6]
```diff
@@ -62,40 +61,57 @@ private void ValidateSecurityModelsDetections(IEnumerable<RegexPattern> patterns
...
 
-                    foreach (var pattern in patterns)
+                    foreach (RegexPattern pattern in patterns)
                     {
                         foreach (string testExample in pattern.GenerateTruePositiveExamples())
                         {
                             string context = testExample;
+                            var matches = CachedDotNetRegex.Instance.Matches(context, pattern.Pattern, captureGroup: "refine");
+                            bool result = matches.Count() == 1;
+                            result.Should().BeTrue(because: $"Pattern {pattern.Id} should match '{context}' exactly once");
```
Comment (rwoll) [inline src/Tests.Microsoft.Security.Utilities.Core/SecretMaskerTests.cs:71]: nit:`matches.Count().Should()…`

Why the extra assignment?

**Review by rwoll — APPROVED** [review]

**Pull request merged**